File: Nancy.ViewEngines.React.UnitTests/ConfigurationTests/DefaultTest.cs
namespace Nancy.ViewEngines.React.UnitTests.ConfigurationTests
{
    using Xunit;

    public class DefaultTest : TestBase
    {
        public DefaultTest()
            : base("Default.config")
        {
        }

        [Fact]
        public void Default_configuration_should_set_up_script_properties()
        {
            Assert.Equal("client", this.Configuration.Script.Dir);
            Assert.Equal("script.js", this.Configuration.Script.Name);
            Assert.Empty(this.Configuration.Script.Extensions);
        }

        [Fact]
        public void Default_configuration_should_set_up_server_assets_path()
        {
            Assert.Equal("assets", this.Configuration.Server.Assets.Path);
        }

        [Fact]
        public void Default_extension_should_be_right_value()
        {
            var defaultExtension = ReactConfiguration.ExtensionElement.DefaultValue;

            Assert.Equal("jsx", defaultExtension.Name);
        }
    }
}

File: Nancy.ViewEngines.React.UnitTests/ConfigurationTests/DefaultTest.js
/* global describe, beforeEach, it */

const path = require('path');
const expect = require('expect.js');
const readFile = require('../../Nancy.ViewEngines.React/tools/commands/file').readFile;
const parseConfig = require('../../Nancy.ViewEngines.React/tools/commands/options').parseConfig;

describe('Default test', () => {
  beforeEach(() => {
    const file = path.resolve(__dirname, '../ConfigurationFixtures', 'Default.config');
    this.parse = readFile(file).then(parseConfig);
  });

  it('should set up script properties', () =>
    this.parse.then(config => {
      expect(config.script.dir).to.be('client');
      expect(config.script.name).to.be('script.js');
      expect(config.script.extensions).to.have.length(1).and.to.contain('jsx');
    }));

  it('should set up default layout', () =>
    this.parse.then(config => {
      const layout = config.script.layout.split(/[\\/]/);

      // TODO should be endsWith
      expect(layout).to.contain('client');
      expect(layout).to.contain('layout.jsx');
    }));

  it('should set up server assets path', () =>
    this.parse.then(config => {
      expect(config.server.assets.path).to.be('assets');
    }));
});
